src/index.ts

```diff
@@ -150,57 +168,78 @@ async function clientAutoMode(r: NginxHTTPRequest): Promise<void> {
     )) as ArrayBuffer
     pkeyPem = toPEM(privKey, 'PRIVATE KEY')
     fs.writeFileSync(pkeyPath, pkeyPem)
-    log.info(`Wrote Private key to ${pkeyPath}`)
+    log.info(`Wrote private key to ${pkeyPath}`)
 
     const challengePath = acmeChallengeDir(r)
 
     try {
       fs.mkdirSync(challengePath, { recursive: true })
     } catch (e) {
-      log.error(
-        `Error creating directory to store challenges. Ensure the ${challengePath} directory is writable by the nginx user.`
-      )
-
-      return r.return(500, 'Cannot create challenge directory')
+      retVal.info.error = `Error creating directory to store challenges. Ensure the ${challengePath} directory is writable by the nginx user.`
+      return retVal
     }
 
     certificatePem = await client.auto({
       csr: Buffer.from(csr.pkcs10Ber),
       email,
       termsOfServiceAgreed: true,
-      challengeCreateFn: async (authz, challenge, keyAuthorization) => {
-        log.info('Challenge Create', { authz, challenge, keyAuthorization })
+      challengeCreateFn: async (_, challenge, keyAuthorization) => {
         log.info(
           `Writing challenge file so nginx can serve it via .well-known/acme-challenge/${challenge.token}`
         )
-        ngx.log(
-          ngx.INFO,
-          `njs-acme: [auto] Writing challenge file so nginx can serve it via ${challengePath}/${challenge.token}`
-        )
         const path = joinPaths(challengePath, challenge.token)
         fs.writeFileSync(path, keyAuthorization)
       },
       challengeRemoveFn: async (_authz, challenge, _keyAuthorization) => {
         const path = joinPaths(challengePath, challenge.token)
         try {
           fs.unlinkSync(path)
-          log.info(`removed challenge ${path}`)
+          log.info(`Removed challenge ${path}`)
         } catch (e) {
-          log.error(`failed to remove challenge ${path}`)
+          log.error(`Failed to remove challenge ${path}`)
         }
       },
     })
     certInfo = await readCertificateInfo(certificatePem)
     fs.writeFileSync(certPath, certificatePem)
-    log.info(`wrote certificate to ${certPath}`)
+    log.info(`Wrote certificate to ${certPath}`)
   }
 
-  const info = {
-    certificate: certInfo,
-    renewedCertificate: renewCertificate,
-  }
+  retVal.success = true
+  retVal.info.certificate = certInfo
+  retVal.info.renewedCertificate = renewCertificate.toString()
```

Comment:
optional nit: Probably would be a little nicer to create this when we need to send it rather than having a long lived mutable thing to worry about.  Seems it's only given a value and returned in two places.